File: frontend/src/types/acknowledgment.ts
export type AcknowledgmentStatus = "draft" | "issued";

export const ACKNOWLEDGMENT_STATUS_LABEL: Record<AcknowledgmentStatus, string> = {
  draft: "下書き",
  issued: "発行済み",
};

export interface AcknowledgmentRead {
  id: string;
  order_id: string;
  project_id: string;
  acknowledgment_number: string | null;
  issue_date: string | null;
  client_address: string | null;
  client_company: string | null;
  client_person: string | null;
  amount_excl_tax: number | null;
  tax_amount: number | null;
  total_amount: number | null;
  stamp_tax: number | null;
  construction_period_start: string | null;
  construction_period_end: string | null;
  payment_condition: string | null;
  terms_and_conditions: string | null;
  status: AcknowledgmentStatus;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/acknowledgment.py
"""注文請書スキーマ。"""
from __future__ import annotations

import uuid
from datetime import date, datetime

from pydantic import BaseModel

from app.models.enums import AcknowledgmentStatus


class AcknowledgmentCreate(BaseModel):
    """注文請書作成スキーマ。注文書からのコピーで生成されるため全フィールドOptional。"""

    issue_date: date | None = None
    client_address: str | None = None
    client_company: str | None = None
    client_person: str | None = None
    amount_excl_tax: float | None = None
    tax_amount: float | None = None
    total_amount: float | None = None
    stamp_tax: float | None = None
    construction_period_start: date | None = None
    construction_period_end: date | None = None
    payment_condition: str | None = None
    terms_and_conditions: str | None = None


class AcknowledgmentUpdate(AcknowledgmentCreate):
    """注文請書更新スキーマ。ステータス変更も可能。"""

    status: AcknowledgmentStatus | None = None


class AcknowledgmentRead(BaseModel):
    """注文請書レスポンス。"""

    id: uuid.UUID
    order_id: uuid.UUID
    project_id: uuid.UUID
    acknowledgment_number: str | None
    issue_date: date | None
    client_address: str | None
    client_company: str | None
    client_person: str | None
    amount_excl_tax: float | None
    tax_amount: float | None
    total_amount: float | None
    stamp_tax: float | None
    construction_period_start: date | None
    construction_period_end: date | None
    payment_condition: str | None
    terms_and_conditions: str | None
    status: AcknowledgmentStatus
    created_at: datetime
    updated_at: datetime
